codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -372,20 +481,45 @@ class ServerProtocolTestGenerator(
         }
     }
 
-    private fun checkParams(httpRequestTestCase: HttpRequestTestCase, rustWriter: RustWriter) {
-        rustWriter.writeInline("let expected = ")
-        instantiator.render(rustWriter, inputShape, httpRequestTestCase.params)
-        rustWriter.write(";")
+    private fun checkRequest(operationShape: OperationShape, operationSymbol: Symbol, httpRequestTestCase: HttpRequestTestCase, rustWriter: RustWriter) {
+        val inputShape = operationShape.inputShape(coreCodegenContext.model)
+        val outputShape = operationShape.outputShape(coreCodegenContext.model)
 
         val operationName = "${operationSymbol.name}${ServerHttpBoundProtocolGenerator.OPERATION_INPUT_WRAPPER_SUFFIX}"
-        rustWriter.rustTemplate(
+        val (inputT, outputT) = operationInputOutputTypes[operationShape]!!
+        val helperNamespace = "crate::operation::protocol_test_helper"
```

Comment:
Consider extracting the string `protocol_test_helper`, since it's used too when defining the module. You also don't need to prefix with `crate::operation::`, since the module is defined in the same file as this chunk of code.